packages/region/src/models/country.ts

```diff
@@ -3,15 +3,20 @@ import {
   Entity,
   ManyToOne,
   OnInit,
+  OptionalProps,
   PrimaryKey,
   Property,
 } from "@mikro-orm/core"
 
 import { generateEntityId } from "@medusajs/utils"
 import Region from "./region"
 
+type CountryOptionalProps = "region" | "region_id"
+
 @Entity({ tableName: "region_country" })
 export default class Country {
+  [OptionalProps]?: CountryOptionalProps
```

Comment:
q: do we need `region/region_id` here since they are marked with optional with `?`